docs/design/proposals/logical-volume-manager-storage-api-v2.md

```diff
@@ -0,0 +1,587 @@
+
+## Motivation
+
+This enhancement proposal presents a comprehensive plan to develop the subsequent version of this API, with the goal of eliminating the limitations originated from assumptions made for single-node development.
+Our aim is to converge our practices towards true best-practice API conventions adhering to upstream standards.
+The proposed changes are designed to facilitate future development by permitting additional non-breaking changes.
+This, in turn, should expedite the iteration process for developers aiming to report extra status updates.
+Furthermore, we aim to simplify the process of debugging and collecting issues when encountering problems with LVMS, which currently lacks a streamlined reporting system. Hence, this proposal ultimately seeks to mitigate such issues and improve the overall user experience.
+
+### User Stories
+
+* As an OpenShift engineer, I want to be able to quickly debug issues with the Logical Volume Manager Storage (LVMS) without having to analyze separate resources for the desired and actual state. This means that the system should clearly display the difference between the expected and current system states, which would allow me to identify and resolve any discrepancies more efficiently.
+* As an OpenShift engineer, I aim to comply with established Kubernetes API Specifications. Adhering to these conventions will simplify the process of onboarding other developers, as they're likely familiar with them, and aligns our company's practices with the wider developer community. This also means I can rely on the well-maintained upstream conventions, avoiding the need to reinvent the wheel.
+* As an OpenShift engineer, I want the flexibility to design non-breaking API changes, allowing a faster iteration period on smaller API changes. This would enable iterative development and a more agile workflow, letting me incrementally build and improve our systems without causing system-wide disruptions.
+* As a user of the LVMS API, I want all pertinent data for Volume Groups and Nodes readily available in one location. Having a centralized place for all related information will improve usability and make the process of managing storage configurations more straightforward and efficient.
+* As a LVMS API user, I want to utilize my preexisting knowledge of Kubernetes APIs to understand the LVMS APIs more quickly. Leveraging knowledge I already have will reduce the learning curve and allow me to work more effectively and efficiently.
+* As an OpenShift cluster administrator, I need to limit the minimum permissions of LVMS to namespace-level as much as possible. By doing so, I can mitigate the risk of privilege escalation into other namespaces, enhancing the overall security of the system. This would help me maintain the integrity and confidentiality of data across different namespaces, providing peace of mind and aiding in regulatory compliance.
```

Comment:
I'm confused what this one has to do with api v2

Reply:
You're right I dont think this is bound to the v2 APIs. I was mainly referring to us probably needing either a fully namespaced or cluster-scoped operator.